src/dataflow-types/src/client/controller.rs

```diff
@@ -153,12 +156,7 @@ where
                         },
                     )
                     .await?;
-                let addrs: Vec<_> = service
-                    .hosts()
-                    .iter()
-                    .map(|h| format!("{h}:2100"))
-                    .collect();
-                let client = RemoteClient::new(&addrs);
+                let client = RemoteClient::new(&service.addresses("controller"));
                 let client: Box<dyn ComputeClient<T>> = Box::new(client);
                 self.compute_mut(instance)
                     .unwrap()
```

Comment:
We probably want to give the instance a better name than `"default"`, but we might not have the infrastructure in place to generate better names.